x/group/simulation/operations.go

```diff
@@ -699,6 +713,133 @@ func SimulateMsgUpdateGroupPolicyMetadata(ak group.AccountKeeper,
+		if err != nil {
+			return simtypes.NoOpMsg(group.ModuleName, TypeMsgWithdrawProposal, ""), nil, err
+		}
+		if account == nil {
+			return simtypes.NoOpMsg(group.ModuleName, TypeMsgWithdrawProposal, "no group member found"), nil, nil
+		}
+
+		spendableCoins := bk.SpendableCoins(sdkCtx, account.GetAddress())
+		fees, err := simtypes.RandomFees(r, sdkCtx, spendableCoins)
+		if err != nil {
+			return simtypes.NoOpMsg(group.ModuleName, TypeMsgWithdrawProposal, "fee error"), nil, err
+		}
+
+		policy := groupPolicy.GetDecisionPolicy()
+		err = policy.Validate(*g)
+		if err != nil {
+			return simtypes.NoOpMsg(group.ModuleName, TypeMsgWithdrawProposal, err.Error()), nil, nil
+		}
+
+		proposalReq, err := group.NewMsgCreateProposalRequest(groupPolicyAddr, []string{acc.Address.String()}, []sdk.Msg{
+			&banktypes.MsgSend{
+				FromAddress: groupPolicyAddr,
+				ToAddress:   accounts[1].Address.String(),
+				Amount:      sdk.Coins{sdk.NewInt64Coin("token", 100)},
+			},
+		}, []byte{}, 0)
+
+		if err != nil {
+			return simtypes.NoOpMsg(group.ModuleName, TypeMsgWithdrawProposal, "fail to create proposal request"), nil, err
+		}
+		_, err = k.CreateProposal(ctx, proposalReq)
+
+		if err != nil {
+			return simtypes.NoOpMsg(group.ModuleName, TypeMsgWithdrawProposal, "failed to create proposal"), nil, err
+		}
+
+		proposalsResult, err := k.ProposalsByGroupPolicy(ctx, &group.QueryProposalsByGroupPolicyRequest{Address: groupPolicyAddr})
+		if err != nil {
+			return simtypes.NoOpMsg(group.ModuleName, TypeMsgWithdrawProposal, "fail to query group info"), nil, err
+		}
+		proposals := proposalsResult.GetProposals()
+		if len(proposals) == 0 {
+			return simtypes.NoOpMsg(group.ModuleName, TypeMsgWithdrawProposal, "no proposals found"), nil, nil
+		}
+
+		var proposal *group.Proposal
+		proposalID := -1
+
+		for _, p := range proposals {
+			if p.Status == group.ProposalStatusSubmitted {
+				timeout := p.Timeout
+				proposal = p
+				proposalID = int(p.ProposalId)
+				if timeout.Before(sdkCtx.BlockTime()) || timeout.Equal(sdkCtx.BlockTime()) {
+					return simtypes.NoOpMsg(group.ModuleName, TypeMsgWithdrawProposal, "voting period ended: skipping"), nil, nil
+				}
+				break
+			}
+		}
+
+		// return no-op if no proposal found
+		if proposalID == -1 {
+			return simtypes.NoOpMsg(group.ModuleName, TypeMsgWithdrawProposal, "no proposals found"), nil, nil
+		}
+
+		// Ensure that group and group policy haven't been modified since the proposal submission.
+		if proposal.GroupPolicyVersion != groupPolicy.Version {
+			return simtypes.NoOpMsg(group.ModuleName, TypeMsgWithdrawProposal, "group policy has been modified"), nil, nil
+		}
+
+		// Ensure the group hasn't been modified.
+		if proposal.GroupVersion != g.Version {
+			return simtypes.NoOpMsg(group.ModuleName, TypeMsgWithdrawProposal, "group has been modified"), nil, nil
+		}
+
+		msg := group.MsgWithdrawProposal{
+			ProposalId: uint64(proposalID),
+			Address:    acc.Address.String(),
```

Comment:
`acc` is a member here but if this member wasn't part of the proposal's proposers then this will fail

Reply:
But I am using `randomMember` helper func to get the group member.

Reply:
this doesn't seem to be fixed, you're still getting a random member which could potentially not be part of the proposers